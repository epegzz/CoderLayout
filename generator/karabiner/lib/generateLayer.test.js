const generateLayer = require('./generateLayer')
const normalizeConfig = require('../../utils/normalizeConfig')

describe('generateLayer', () => {
  it('works', () => {
    const config = normalizeConfig({
      layers: {
        test: {
          description: 'description',
          trigger: ['x', 'y'],
          mappings: [{ a: 'b'}, {c: 'd'}]
        },
        test2: {
          description: 'description2',
          trigger: 'x',
          mappings: [{ a: 'b'}, {c: 'd'}]
        },
      },
    })
    const result = generateLayer('test', config.layers.test)
    const result2 = generateLayer('test', config.layers.test2)
    expect(result).toMatchSnapshot()
    expect(result2).toMatchSnapshot()
  })
})
