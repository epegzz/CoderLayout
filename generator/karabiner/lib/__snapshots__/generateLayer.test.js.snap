// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateLayer works 1`] = `
Object {
  "description": "description",
  "manipulators": Array [
    Object {
      "conditions": Array [
        Object {
          "name": "trigger_x",
          "type": "variable_if",
          "value": 1,
        },
        Object {
          "name": "trigger_y",
          "type": "variable_if",
          "value": 1,
        },
      ],
      "from": Object {
        "key_code": "a",
        "modifiers": Object {
          "mandatory": Array [],
          "optional": Array [
            "any",
          ],
        },
      },
      "to": Array [
        Object {
          "key_code": "b",
        },
      ],
      "type": "basic",
    },
    Object {
      "conditions": Array [
        Object {
          "name": "trigger_x",
          "type": "variable_if",
          "value": 1,
        },
        Object {
          "name": "trigger_y",
          "type": "variable_if",
          "value": 1,
        },
      ],
      "from": Object {
        "key_code": "c",
        "modifiers": Object {
          "mandatory": Array [],
          "optional": Array [
            "any",
          ],
        },
      },
      "to": Array [
        Object {
          "key_code": "d",
        },
      ],
      "type": "basic",
    },
  ],
}
`;

exports[`generateLayer works 2`] = `
Object {
  "description": "description2",
  "manipulators": Array [
    Object {
      "conditions": Array [
        Object {
          "name": "trigger_x",
          "type": "variable_if",
          "value": 1,
        },
      ],
      "from": Object {
        "key_code": "a",
        "modifiers": Object {
          "mandatory": Array [],
          "optional": Array [
            "any",
          ],
        },
      },
      "to": Array [
        Object {
          "key_code": "b",
        },
      ],
      "type": "basic",
    },
    Object {
      "conditions": Array [
        Object {
          "name": "trigger_x",
          "type": "variable_if",
          "value": 1,
        },
      ],
      "from": Object {
        "key_code": "c",
        "modifiers": Object {
          "mandatory": Array [],
          "optional": Array [
            "any",
          ],
        },
      },
      "to": Array [
        Object {
          "key_code": "d",
        },
      ],
      "type": "basic",
    },
  ],
}
`;
